src/Components/Components/src/PersistentState/ComponentStatePersistenceManager.cs

```diff
@@ -55,9 +55,33 @@ public ComponentStatePersistenceManager(ILogger<ComponentStatePersistenceManager
     /// <param name="store">The <see cref="IPersistentComponentStateStore"/> to restore the application state from.</param>
     /// <returns>A <see cref="Task"/> that will complete when the state has been restored.</returns>
     public async Task RestoreStateAsync(IPersistentComponentStateStore store)
+    {
+        await RestoreStateAsync(store, scenario: null);
+    }
+
+    /// <summary>
+    /// Restores component state from the given store with scenario context.
+    /// </summary>
+    /// <param name="store">The store to restore state from.</param>
+    /// <param name="scenario">The restoration scenario context.</param>
+    /// <returns>A task that completes when state restoration is finished.</returns>
+    public async Task RestoreStateAsync(
+        IPersistentComponentStateStore store, 
+        IPersistentComponentStateScenario? scenario)
     {
         var data = await store.GetPersistedStateAsync();
-        State.InitializeExistingState(data);
+        
+        if (scenario == null)
+        {
+            // First-time initialization
+            State.InitializeExistingState(data);
+        }
+        else
+        {
+            // Scenario-based update
+            State.UpdateExistingState(data, scenario);
+        }
```

Comment:
No, you should track the number of times you get called. not whether you receive an scenario or not